addons/payment/tests/test_payment_token.py

```diff
@@ -31,6 +31,8 @@ def test_cannot_assign_token_to_public_partner(self):
     def test_token_cannot_be_unarchived(self):
         """ Test that unarchiving disabled tokens is forbidden. """
         token = self._create_token(active=False)
+        token.payment_method_id.active = False
+        token.provider_id.state='disabled'
```

Comment:
May be add other assertion with provider enable or not disable and other with method active or not

Reply:
Hello @kcv-odoo ,
I have added assertion with payment provider and method.

Thankyou :smile: